dali/ft/filecopy.cpp

```diff
@@ -2596,6 +2625,190 @@ void FileSprayer::pushParts()
     performTransfer();
 }
 
+static StringBuffer & buildAzureStorageURIBase(IStoragePlane * storagePlane, StringBuffer & uri)
+{
+    const char *accountName = storagePlane->queryStorageApiAccount();
+    uri.appendf("https://%s", accountName);
+
+    if (storagePlane->getStorageType()==StorageType::StorageTypeAzureFile)
+        uri.append(".file");
+    else
+        uri.append(".blob");
+    uri.appendf(".core.windows.net/%s", storagePlane->queryStorageContainer());
+
+    return uri;
+}
+
+using namespace Azure::Storage;
+using namespace Azure::Storage::Files;
+using namespace Azure::Storage::Blobs;
+
+bool FileSprayer::transferUsingAPI()
```

Comment:
minor: Azure specific aspects of this should be encapsulated in some azure specific classes.  Ok to do as a separate PR, as long as the jira is created.